agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java

```diff
@@ -889,17 +885,13 @@ static boolean applyCommonTags(
 
   static boolean applyConnectionStringAndRoleNameOverrides(
       AbstractTelemetryBuilder telemetryBuilder, Object value, String key) {
-    if (key.equals(AiSemanticAttributes.CONNECTION_STRING.getKey()) && value instanceof String) {
+    if (key.equals(AiSemanticAttributes.INTERNAL_CONNECTION_STRING.getKey())
+        && value instanceof String) {
       // intentionally letting exceptions from parse bubble up
       telemetryBuilder.setConnectionString(ConnectionString.parse((String) value));
       return true;
     }
-    if (key.equals(AiSemanticAttributes.INSTRUMENTATION_KEY.getKey()) && value instanceof String) {
-      // intentionally letting exceptions from parse bubble up
-      telemetryBuilder.setConnectionString(ConnectionString.parse("InstrumentationKey=" + value));
-      return true;
-    }
-    if (key.equals(AiSemanticAttributes.ROLE_NAME.getKey()) && value instanceof String) {
+    if (key.equals(AiSemanticAttributes.INTERNAL_ROLE_NAME.getKey()) && value instanceof String) {
```

Comment:
This pattern seems to appear several times. The code may be a bit shorter and easier to read with the addition of a `hasStringValue` method (perhaps in another PR).

Reply:
or don't need to check value type. each key clearly defines the value type.

Reply:
I created work item to revisit this sometime after the release: https://msazure.visualstudio.com/One/_workitems/edit/15574583